tests/unit/lms/services/rsa_key_service_test.py

```diff
@@ -90,6 +136,12 @@ def jwk(self, patch):
     def rsa(self, patch):
         return patch("lms.services.rsa_key.rsa")
 
+    @pytest.fixture
+    def svc_generates(self, svc):
+        with patch.object(svc, "generate") as generate:
+            generate.side_effect = lambda: factories.RSAKey(expired=False)
+            yield svc
```

Comment:
:thinking: There's something fishy about this. In general we don't need to mock out things like this.

When I remove this fixture (by making it return `svc`), everything bar one test still passes, complaining about converting `MagicMock`s to binary, but I can't see where a mock is getting in.

What's this all about?

Reply:
I think I've worked it out. Our `AESService` mock isn't returning strings, so we can't use it with the DB, which means we disable this.

If we have something like this:

```python
    @pytest.fixture
    def aes_service(self, aes_service):
        # Don't return MagicMock objects from these functions so the output can
        # be tested against things which store them in the DB
        aes_service.encrypt.side_effect = lambda aes_iv, plain_text: plain_text
        aes_service.build_iv.return_value = b"fake_aes_iv"
        return aes_service
```

It's no longer required. I wonder if this should be in the base service fixture?